### ISSUE TITLE
Question About Data Preprocessing and Rendering Artifacts

### ISSUE THREAD

**Issue by Selena1105** [Beerend/NeRF-OR#2]
Hi！

Thank you again for sharing your work . I have a couple of new questions regarding my experiments.

### What I Did:
1. I input the original-size colorimage from the 4D-OR dataset into the Marigold model and used the marigold-v1-0 pretrained weights to obtain .npy dense depth estimations.
2. I converted the depth maps into .tiff format using the tifffile library’s tiff.imwrite() function.
3. I downsampled the colorimage, sensor depth, and depth estimation by 4x using a max-pooling method to prepare the final dataset.
4. I used the experimental parameters in ./configs/4D-OR/static/take1_frame_074.txt without modifications (e.g., image_scale=4 and N_iters=2000).

### My Questions:
1. Is my preprocessing pipeline consistent with your approach?
2. I noticed that in the final rendered results, certain viewpoints appear very blurry, and some scenes are obstructed by large fog-like artifacts, as shown in the attached images. Do you know what might cause this issue, or could you suggest any improvements to address it?
3. Why normaliztion to [0,1] is required for RGB images in the load_4dor.py?

Thank you very much for your time and support !!

![Image](https://github.com/user-attachments/assets/c8bd2321-9d72-4079-8991-9d884bed60db)

![Image](https://github.com/user-attachments/assets/e8eac467-2ea2-42ef-971f-517ff2dddeef)

**Comment by Beerend** [comment]
Hi @Selena1105,

Thank you for those questions, and apologies for the late reply. I uploaded two scripts [prepare_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/prepare_4DOR.py) and [postprocess_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/postprocess_4DOR.py), which I used for processing the data before and after running marigold. The scripts are not well documented, but I hope they are of help!

1. Considering the data processing, I found a couple of things to be important:
- When resizing depths, the images need to be resampled with nearest neighbor interpolation (```cv2.resize(img, (x, y), interpolation=cv2.INTER_NEAREST')''). This avoids wrongly interpolated depth values at the edges of objects.
- It helps to scale the estimated depth images to the same range of values as the depth images, e.g., using least-squares.
- The estimated depth images should represent depth, not disparity, meaning that close objects should have low values and far objects should have large values.

2. To address to blurry or foggy renders, there is certainly a limit to the camera positions that can be rendered accurately. I saw that using depth and normals from estimated depth helps in reducing the fog, but the further the render camera position is from the training positions the worse the results get.

3. The RGB values are scaled between 0 and 1 because of the loss function that compares the rendered pixels with the ground truth pixels.

I hope this could help you a bit further!

**Comment by Selena1105** [comment]
> Hi [@Selena1105](https://github.com/Selena1105),
> 
> Thank you for those questions, and apologies for the late reply. I uploaded two scripts [prepare_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/prepare_4DOR.py) and [postprocess_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/postprocess_4DOR.py), which I used for processing the data before and after running marigold. The scripts are not well documented, but I hope they are of help!
> 
> 1. Considering the data processing, I found a couple of things to be important:
> 
> * When resizing depths, the images need to be resampled with nearest neighbor interpolation (```cv2.resize(img, (x, y), interpolation=cv2.INTER_NEAREST')''). This avoids wrongly interpolated depth values at the edges of objects.
> * It helps to scale the estimated depth images to the same range of values as the depth images, e.g., using least-squares.
> * The estimated depth images should represent depth, not disparity, meaning that close objects should have low values and far objects should have large values.
> 
> 2. To address to blurry or foggy renders, there is certainly a limit to the camera positions that can be rendered accurately. I saw that using depth and normals from estimated depth helps in reducing the fog, but the further the render camera position is from the training positions the worse the results get.
> 3. The RGB values are scaled between 0 and 1 because of the loss function that compares the rendered pixels with the ground truth pixels.
> 
> I hope this could help you a bit further!



> Hi [@Selena1105](https://github.com/Selena1105),
> 
> Thank you for those questions, and apologies for the late reply. I uploaded two scripts [prepare_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/prepare_4DOR.py) and [postprocess_4DOR.py](https://github.com/Beerend/NeRF-OR/blob/main/marigold/postprocess_4DOR.py), which I used for processing the data before and after running marigold. The scripts are not well documented, but I hope they are of help!
> 
> 1. Considering the data processing, I found a couple of things to be important:
> 
> * When resizing depths, the images need to be resampled with nearest neighbor interpolation (```cv2.resize(img, (x, y), interpolation=cv2.INTER_NEAREST')''). This avoids wrongly interpolated depth values at the edges of objects.
> * It helps to scale the estimated depth images to the same range of values as the depth images, e.g., using least-squares.
> * The estimated depth images should represent depth, not disparity, meaning that close objects should have low values and far objects should have large values.
> 
> 2. To address to blurry or foggy renders, there is certainly a limit to the camera positions that can be rendered accurately. I saw that using depth and normals from estimated depth helps in reducing the fog, but the further the render camera position is from the training positions the worse the results get.
> 3. The RGB values are scaled between 0 and 1 because of the loss function that compares the rendered pixels with the ground truth pixels.
> 
> I hope this could help you a bit further!

Thank you for your reply ! It's really helpful.
